crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -423,3 +410,75 @@ fn is_in_async_function(node: &JsExpressionStatement) -> bool {
         })
         .is_some()
 }
+
+/// Checks if the initializer of a `JsVariableDeclarator` is an async function.
+///
+/// Example TypeScript code that would return `true`:
+///
+/// ```typescript
+/// const returnsPromise = async (): Promise<string> => {
+///   return 'value';
+/// }
+///
+/// const returnsPromise = async function (): Promise<string> {
+///   return 'value'
+/// }
+/// ```
+fn is_variable_initializer_a_promise(js_variable_declarator: &JsVariableDeclarator) -> bool {
+    let Some(initializer_clause) = &js_variable_declarator.initializer() else {
+        return false;
+    };
+    let Ok(expr) = initializer_clause.expression() else {
+        return false;
+    };
+    match expr {
+        AnyJsExpression::JsArrowFunctionExpression(arrow_func) => {
+            arrow_func.async_token().is_some()
+                || is_return_type_a_promise(arrow_func.return_type_annotation())
+        }
+        AnyJsExpression::JsFunctionExpression(func_expr) => {
+            func_expr.async_token().is_some()
+                || is_return_type_a_promise(func_expr.return_type_annotation())
+        }
+        _ => false,
+    }
+}
+
+/// Checks if a `JsVariableDeclarator` has a TypeScript type annotation of `Promise`.
+///
+///
+/// Example TypeScript code that would return `true`:
+/// ```typescript
+/// const returnsPromise: () => Promise<string> = () => {
+///   return Promise.resolve("value")
+/// }
+/// ```
+fn is_variable_annotation_a_promise(js_variable_declarator: &JsVariableDeclarator) -> bool {
```

Comment:
Same here. Change it to `Option<bool>`, use `.ok()?` every time you extract a type from a node, and  the caller will use `?`